airflow/config_templates/config.yml

```diff
@@ -944,6 +944,14 @@ metrics:
   description: |
     StatsD (https://github.com/etsy/statsd) integration settings.
   options:
+    metrics_use_fuzzy_match:
```

Comment:
I am personally against adding a new config for `metrics_use_fuzzy_match`. With this approach, we will keep sprawling the number of configurations every time we need to make a hard decision. I am of the opinion that we should have just 2 configs - `metrics_allow_list` and `metrics_block_list` - and they should be changed to support the pattern matching. That means, we should deprecate existing configs now and introduce new ones which are more powerful and support pattern matching, in order to keep it backward compatible. Yes, this means we need to go through deprecation until v3.0, but I think that is better for the product.

Reply:
I would also suggest renaming this to `metrics_enable_pattern_match` or `metrics_use_pattern_match`, in case we really decide to go with this.

Reply:
I like the idea of pattern match, we even don't need to rename the variable, we can just update the configuration description and explain that since 2.9.0, Airflow supports regex in `metrics_allow_list` and `metrics_block_list`.

Reply:
That feels like a breaking change to me, no?  What previously only matched the start of the metric name string will now match any substring in the name.

Reply:
> then likely a bunch of confusion when the default changes after that.

This is the primary reason I think moving over to two new configs that replace old configs as better. This way when the old config is deprecated, users who adopted the new configs wouldn't have to make any change at all.

Reply:
>This is the primary reason I think moving over to two new configs that replace old configs as better. This way when the old config is deprecated, users who adopted the new configs wouldn't have to make any change at all.

But now isn't this adding even more new configs than Dennis's original proposal?

Ultimately, I agree that we should change the definition of the current configs and add nothing new (zero new configs added) but we also do need _some_ way to give people fair warning. I think we should try do that with adding as few, if any, configs as possible. So I'm in slight favour of Dennis's proposal since it only adds one new config, which will be dropped after the deprecation process/switch over. Another approach could be a version check in the code that says something like "the behaviour of these configs will change by release <foo>, please be prepared to update your configuration at that time" (or a better message since I just spit balled that right now...) and uses the old beahviour if the code is before the release where we switch, and later releases it uses the new behaviour.

Reply:
It looks like there are a number of "less-bad" solutions but no real good one.   Does anyone feel particularly strongly about any of the options here?

If nobody has a strong opinion, I'm going to add "if metrics_use_fuzzy_match is false, display a deprecation notice" and leave it at that for now.

Reply:
I don't feel _too_ strongly, not willing to die on any hills here.

Reply:
Based on discussion here, I'm going to add a deprecation warning and rename "fuzzy matching" to "pattern matching".